Source/WebGPU/WebGPU/WebGPUSwift.h

```diff
@@ -25,11 +25,16 @@
 
 #pragma once
 
-#ifdef __cplusplus
-#include <span>
-#include <stdint.h>
-
-using SpanConstUInt8 = std::span<const uint8_t>;
-using SpanUInt8 = std::span<uint8_t>;
-
+// Current Swift-C++ encapsulation rules prevent Swift from accessing non-public data members,
+// even in extensions. When building WebGPU, use these macros to allow our Swift module to break
+// encapsulation.
+#if defined(__swift__) && __swift__ && \
+    defined(__WEBGPU__) && __WEBGPU__ && \
+    defined(ENABLE_WEBGPU_SWIFT) && ENABLE_WEBGPU_SWIFT
+#define WebGPUSwift_protected   public
```

Comment:
I'm struggling with this naming because it tells the reader where it will be used, and not what it's going to do.

We could make it more verbose:
`PROTECTED_BUT_PUBLIC_IN_WEBGPU_SWIFT:`

Or use it like this:
```
protected:
PUBLIC_IN_WEBGPU_SWIFT:

```
Or like this:

`protected: PUBLIC_IN_WEBGPU_SWIFT:`

I think I like the last version best? It's a bit weird, but it says what it does.

Reply:
That makes sense, you've convinced me too. Though I think it needs to be

`protected: PUBLIC_IN_WEBGPU_SWIFT`

(no trailing colon) so that the macro can expand to nothing outside of the Swift target's compilation.